subprojects/docs/src/docs/userguide/console.adoc

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[console]]
+== The Gradle Console
+
+NOTE: This feature is a work in progress.
+
+[[sec:console_overview]]
+=== Overview
+
+The console is the central user interface for any user executing Gradle from the command line. Gradle's console is optimized for rendering performance, showing only relevant information while at the same time providing a visually appealing feedback loop.
+
++++++
+<figure xmlns:xi="http://www.w3.org/2001/XInclude">
+    <title>The Gradle console in action</title>
+    <imageobject>
+        <imagedata fileref="img/console-animation.gif"/>
+    </imageobject>
+</figure>
++++++
+
+=== Visual elements
+
+The visual representation of the console is broken up into various elements with the goal of being able to concentrate on the information of interest. Each of these visual elements help with answering specific questions while executing a Gradle build.
+
+- <<sec:console_build_progress_bar,The build progress bar>>: When will my build finish?
+- <<sec:console_work_in_progress_display,The work in-progress display>>: What is Gradle doing right now?
+- <<sec:console_build_result,The build result>>: Are there other interesting outcomes e.g. tasks that have been skipped or were up-to-date?
+- <<sec:console_build_output,The build output>>: Is there anything I should know about my build right now e.g. tests have failed and emitted warnings?
```

Comment:
Are these sorted in any particular order?

I think it would make sense to list these in the order they appear:
1. build output
1. build progress bar
1. work in progress
1. build result